server/src/main/java/org/candlepin/auth/ConsumerAuth.java

```diff
@@ -39,41 +35,37 @@
     protected ConsumerCurator consumerCurator;
     protected OwnerCurator ownerCurator;
     protected DeletedConsumerCurator deletedConsumerCurator;
-    private Provider<I18n> i18nProvider;
 
     @Inject
     ConsumerAuth(ConsumerCurator consumerCurator, OwnerCurator ownerCurator,
-        DeletedConsumerCurator deletedConsumerCurator,
-        Provider<I18n> i18nProvider) {
+        DeletedConsumerCurator deletedConsumerCurator) {
         this.consumerCurator = consumerCurator;
         this.ownerCurator = ownerCurator;
         this.deletedConsumerCurator = deletedConsumerCurator;
-        this.i18nProvider = i18nProvider;
     }
 
+    /**
+     * Creates a principal according to the {@link Consumer} with the given
+     * consumerUuid.
+     *
+     * @param consumerUuid requested consumer
+     * @return created principal
+     */
     public ConsumerPrincipal createPrincipal(String consumerUuid) {
-        ConsumerPrincipal principal = null;
-
-        if (consumerUuid != null) {
-            // If this UUID has been deleted, return a 410.
```

Comment:
While I'm very much in favor of this being less transparent for data security reasons, I'd be interested to know if there's an old, explicit reason we returned a 410 in this case. Given that it's more work to check for the consumer's state here, I have to assume someone requested it at one point.

Reply:
I wrote this bit 7 years ago and unfortunately, I can't remember anything about it.  However, looking at [subscription-manager](https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/scripts/rhsmcertd_worker.py#L128), I do think we want to continue returning 410 if the consumer has been deleted.

```python
if ge.deleted_id == uuid:
    log.critical("Consumer profile \"%s\" has been deleted from the server. Its local certificates will now be archived", uuid)
    managerlib.clean_all_data()
    log.critical("Certificates archived to '/etc/pki/consumer.old'. Contact your system administrator if you need more           information.")
```

Reply:
Done. I reverted the changes so that the return code stays 410.